### PR TITLE
Split read/writes in a cluster

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Adds support for splitting reads and writes to different databases in a cluster
Containerizes haproxy-admin

# Impacts
L3 clusters
haproxy-admin

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [X] Document the feature
- [N/A] Add unit tests
- [X] Add acceptance tests (TestLink) -> Configure a layer 3 cluster with split read and writes

# NEWS file entries
## New Features
* Added support to split database read and writes to different MySQL servers

### CODE REVIEW

**Comment by julsemaan** [comment]
Let's review + merge this ASAP, I thoroughly tested the code from @fdurand and all adjustments have been made so that it does what it should

File: docs/cluster/layer_3_clusters.asciidoc [hunk 1]
```diff
@@ -496,3 +496,111 @@ The replication MASTER_USER and MASTER_PASSWORD can be found in the main sites p
...
+type=management
+
+[DC2 pf1-dc2.example.com interface ens224]
+ip=192.168.22.11
+mask=255.255.255.0
+enforcement=vlan
+type=internal
+----
+
+In this example, DC1 will send its read and writes to the primary database and DC2 will send its writes to the primary database and its reads to the replica located in DC2. You can have multiple comma-delimited read replicas defined in `db_read` which will be used in priority in the order they are defined.
```
Comment (extrafu) [inline docs/cluster/layer_3_clusters.asciidoc]: its reads and writes

**Comment by extrafu** [comment]
Aside from the small typo, all good for me.

**Comment by julsemaan** [comment]
Fixed the typo, @nqb or @JeGoi, can you take a look at this just to make sure we have another external pair of eyes looking at the code and then we'll merge if it's all good

File: conf/proxysql.conf.example [hunk 2]
```diff
@@ -99,6 +115,13 @@ mysql_query_rules =
...
     },
+    {
+        rule_id=200
+        active=1
+        match_pattern=".*read_only.*"
+        destination_hostgroup=10
+        apply=1
+    },
     {
         rule_id=200
```
Comment (JeGoi) [inline conf/proxysql.conf.example]: Isn't it an issue to have two rule_id with the same number?

File: docs/cluster/assumptions.asciidoc [hunk 3]
```diff
@@ -19,9 +19,11 @@ endif::[]
 * You have at least three (3) installed PacketFence (v7+) servers
 * The servers are running one of RHEL 8 / Debian 11
 * The servers have identical identifiers for network interfaces (e.g. eth0) (see next section)
-* The servers network interfaces are on the same layer 2 network
 * The servers must have IPv6 disabled (see next section)
 * The servers must have a fully qualified domain name (FQDN) to identify them
+* The servers are located within 5ms of latency of each other (requirement for Galera cluster)
+ * For smaller deployments, 10ms of latency can be tolerated
+ * PacketFence does support exteral MySQL primary/replicas to allow deployments which are geo distributed (see 'Geo Distributed Database' in this document for details)
```
Comment (JeGoi) [inline docs/cluster/assumptions.asciidoc]:
```suggestion
 * PacketFence does support external MySQL primary/replicas to allow deployments which are geo distributed (see 'Geo Distributed Database' in this document for details)
```

**Review by JeGoi — COMMENTED** [review]
Just 1 question and one proposition

**Review by JeGoi — APPROVED** [review]
Cool, looks great to me!

**Pull request merged**